internal/services/storage/storage_share_file_resource.go

```diff
@@ -84,6 +85,12 @@ func resourceStorageShareFile() *pluginsdk.Resource {
 				ForceNew:     true,
 			},
 
+			"content_length": {
+				Type:     pluginsdk.TypeInt,
+				Default:  nil,
```

Comment:
should this be 0?

Reply:
due to its a computed schema, the default value has to be `nil` and should be set at creation time.
As more I think about that, I think I should move the d.Set("content_length", int(*props.ContentLength)) into the create function directly (like the documentation state for the `Computed` property: `If Computed is true, then the result of this value is computed
// (unless specified by config) on creation.` and not on every read)

Reply:
Thanks for explaining @fwiedmann - in this case then we should probably remove default value altogether since this is never set in the config. 

This documentation means that the property will receive a value on creation - it is correct to set this value in the Read method 😄

Reply:
Also, if content_length is never set in the config, we probably don't need to set this in the Update method either

Reply:
Default value is removed and the set of the content_length is now in the create method.
We need to send the content length in the update request, or the client package will set the value to 0 (default int value). This is actually the fix for the bug that caused that the file content gets erased after an update request 😄

Reply:
Sorry for the confusion @fwiedmann - content_length should be set in the **Read** method. 

Once this change is done I will run the tests and see about merging this 🚀